evadb/binder/statement_binder.py

```diff
@@ -145,7 +147,34 @@ def _bind_create_index_statement(self, node: CreateIndexStatement):
         assert len(node.col_list) == 1, "Index cannot be created on more than 1 column"
 
         # TODO: create index currently only works on TableInfo, but will extend later.
-        assert node.table_ref.is_table_atom(), "Index can only be created on Tableinfo"
+        assert (
+            node.table_ref.is_table_atom()
+        ), "Index can only be created on an existing table"
+
+        # Vector type specific check.
+        catalog = self._catalog()
```

Comment:
Reminder to move it outside

Reply:
@gaurav274 Added some changes. Can you give it a read? Not sure shall we still stick with the `singledispatchmethod` approach or do subclass inheritance and add some if statement like the `Executor`.